index.html

```diff
@@ -86,51 +106,216 @@
 <script type="text/javascript" src="https://npmcdn.com/d3@3.5.17/d3.min.js"></script>
 <script type="text/javascript" src="https://npmcdn.com/science@1.9.3/science.v1.min.js"></script>
 <script type="text/javascript" src="https://npmcdn.com/simple-statistics@2.0.0/dist/simple-statistics.min.js"></script>
+<script type="text/javascript" src="https://npmcdn.com/d3-tip@0.6.7/index.js"></script>
+<script type="text/javascript" src="https://npmcdn.com/d3-tip@0.6.7/index.js"></script> 
 <script>
   var data = [];
   var options = {
-    yearRange: 'all'
+    party: 'all'
   };
 
-  var svg;
-
+  var margin = { top: 15, right: 15, bottom: 30, left:45 };
+  var width = 640 - margin.right -  margin.left;
+  var height = 400 - margin.top - margin.bottom;
   // FETCH DATA
+  d3.json('data.json', function (error, json){
+    // Call back function
+    if (error) { throw error; }
+    data = json;
 
-  function setup() {
-    // SVG
+    var charts = [
+      new Chart('#chart')
+    ];
 
-    svg = d3.select('#chart')
-      .append('svg');
+    // EVENT HANDLERS
+    d3.select('#party').on('change', function () {
+      options.party = d3.event.target.value;
+      charts.forEach(function (chart) {chart.update();});
+    });
+  });
 
+  function Chart(selector, variable, title) {
+    var chart = this;
+
+    chart.variable = variable;
+    
+    // SVG
+    chart.svg = d3.select(selector)
+      .append('svg')
+      .attr('width', width + margin.right + margin.left)
+      .attr('height', height + margin.top + margin.bottom)
+      .append('g')
+      .attr('transform','translate(' + margin.left + ',' + margin.top + ')');
+   
     // SCALES
+    chart.x = d3.scale.linear()
+      .domain([d3.min(data,function (d){ return d.cfScore;}), d3.max(data, function (d){ return d.cfScore; } ) ])
+      .range([margin.left,width])
+      .nice();
+
+    chart.y = d3.scale.linear()
+      .domain(d3.extent(data, function (d){ return d.distPart; } ))
+      .range([(height-margin.top),0])
+      .nice();    
 
     // AXES
+    var xAxis = d3.svg.axis()
+      .scale(chart.x)
+      .tickSize(-height)
+      .orient("bottom");
+    var yAxis = d3.svg.axis()
+      .scale(chart.y)
+      .tickSize(-width)
+      .orient("left");
 
-    // EVENT HANDLERS
+    chart.svg.append("g")
+      .classed("x axis", true)
+      .attr("transform", "translate(0," + (height-margin.top) + ")")
+      .call(xAxis);
+    chart.svg.append("g")
+      .classed("y axis", true)
+      .attr("transform","translate(" + margin.left + ",0)")
+      .call(yAxis);
+    var line = d3.svg.line()
+      .x(function(d) { return xScale(d.x); })
+      .y(function(d) { return yScale(d.y); });
```

Comment:
Last night I was trying to figure out why I have that line in my code... I see now that the variable only exists in the setup function. I'll correct this as soon as I can.